patches/server/0749-Configurable-sculk-sensor-listener-range.patch

```diff
@@ -0,0 +1,60 @@
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Thu, 19 Aug 2021 18:45:42 -0700
+Subject: [PATCH] Configurable sculk sensor listener range
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/Blocks.java b/src/main/java/net/minecraft/world/level/block/Blocks.java
+index af4287e2fff8bc920f615ba56f78bd5acdec8721..a3b1b566f1c5a5168eb665bf7ea26afac40c6cf3 100644
+--- a/src/main/java/net/minecraft/world/level/block/Blocks.java
++++ b/src/main/java/net/minecraft/world/level/block/Blocks.java
+@@ -951,7 +951,7 @@ public class Blocks {
+         return 1;
+     }).emissiveRendering((state, world, pos) -> {
+         return SculkSensorBlock.getPhase(state) == SculkSensorPhase.ACTIVE;
+-    }), 8));
++    }), 8)); // Paper - diff on change
+     public static final Block OXIDIZED_COPPER = register("oxidized_copper", new WeatheringCopperFullBlock(WeatheringCopper.WeatherState.OXIDIZED, BlockBehaviour.Properties.of(Material.METAL, MaterialColor.WARPED_NYLIUM).requiresCorrectToolForDrops().strength(3.0F, 6.0F).sound(SoundType.COPPER)));
+     public static final Block WEATHERED_COPPER = register("weathered_copper", new WeatheringCopperFullBlock(WeatheringCopper.WeatherState.WEATHERED, BlockBehaviour.Properties.of(Material.METAL, MaterialColor.WARPED_STEM).requiresCorrectToolForDrops().strength(3.0F, 6.0F).sound(SoundType.COPPER)));
+     public static final Block EXPOSED_COPPER = register("exposed_copper", new WeatheringCopperFullBlock(WeatheringCopper.WeatherState.EXPOSED, BlockBehaviour.Properties.of(Material.METAL, MaterialColor.TERRACOTTA_LIGHT_GRAY).requiresCorrectToolForDrops().strength(3.0F, 6.0F).sound(SoundType.COPPER)));
+diff --git a/src/main/java/net/minecraft/world/level/block/entity/SculkSensorBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/SculkSensorBlockEntity.java
+index 2b0c3d26feb2c34bb5762f8a12f4cab31b7dd650..c8219c2c63ab0fbdde5c3da046f7f2312ced895b 100644
+--- a/src/main/java/net/minecraft/world/level/block/entity/SculkSensorBlockEntity.java
++++ b/src/main/java/net/minecraft/world/level/block/entity/SculkSensorBlockEntity.java
+@@ -26,12 +26,15 @@ public class SculkSensorBlockEntity extends BlockEntity implements VibrationList
+     public void load(CompoundTag nbt) {
+         super.load(nbt);
+         this.lastVibrationFrequency = nbt.getInt("last_vibration_frequency");
++        if (nbt.contains(LISTENER_RANGE_NBT_KEY)) this.listener.listenerRange = nbt.getInt(LISTENER_RANGE_NBT_KEY); // Paper
+     }
+ 
++    private static final String LISTENER_RANGE_NBT_KEY = "Paper.ListenerRange"; // Paper
+     @Override
+     public CompoundTag save(CompoundTag nbt) {
+         super.save(nbt);
+         nbt.putInt("last_vibration_frequency", this.lastVibrationFrequency);
++        if (this.listener.listenerRange != 8) nbt.putInt(LISTENER_RANGE_NBT_KEY, this.listener.listenerRange); // Paper - 8 is default listener range defined by Blocks#SCULK_SENSOR
```

Comment:
An alternative to the hardcoded value would be this:
```suggestion
+        if (this.listener.listenerRange != ((SculkSensorBlock) Blocks.SCULK_SENSOR).getListenerRange()) nbt.putInt(LISTENER_RANGE_NBT_KEY, this.listener.listenerRange); // Paper - 8 is default listener range defined by Blocks#SCULK_SENSOR
```

In regards to the casting, I can see tho how this might not be the cleanest solution either, I just wanted to at least write it down so people can throw their opinion onto it.

Reply:
Yeah, I think even with the casting its probably better to do it that way. Can remove the diff on change comment as well with your way.